components/util/Anchors.tsx

```diff
@@ -73,9 +87,62 @@ export function AnchorsSetter(props) {
 
   useEffect(() => {
     setAnchors(anchors);
+    setActiveAnchor(anchors[0]);
 
-    return () => setAnchors([]);
+    return () => {
+      setAnchors([]);
+      setActiveAnchor(null);
+    };
   }, [router?.pathname]);
 
+  useLayoutEffect(() => {
+    let raf: number;
```

Comment:
`raf` will be used to store a reference (number) to a `requestAnimationFrame` which we can use to cancel it later if we need to.